aeneas/src/v3/TypeSystem.v3

```diff
@@ -59,6 +59,18 @@ component TypeSystem {
 				}
 				return false;
 			}
+			V3Kind.REF => {
+				if (t1 == Null.TYPE) return false;
+				if (t1.typeCon.kind != V3Kind.REF) return false;
+				var reft1 = V3Ref_TypeCon.!(t1.typeCon);
```

Comment:
The more robust way to do this is to implement `getSuperType()` for struct ref types. That should return `ref<T>` for `Ref<T>` types and `null` for everything else. Then use a similar style of loop from the CLASS case above.